src/uv_os.c

```diff
@@ -37,6 +39,13 @@ int UvOsClose(uv_file fd)
     return uv_fs_close(NULL, &req, fd, NULL);
 }
 
+#ifdef _WIN32
+int UvOsFallocate(uv_file fd, off_t offset, off_t len)
+{
+    //TODO: implement functionality
+    return 0;
+}
+#else
```

Comment:
It seems like `posix_fallocate` is also missing on OSX, I found out that mozilla is faking it by using `ftruncate` under the hood.

https://stackoverflow.com/a/11497568/1941280